setupext.py

```diff
@@ -942,15 +971,66 @@ def version_from_header(self):
                 return '.'.join([major, minor, patch])
 
     def add_flags(self, ext):
-        pkg_config.setup_extension(
-            ext, 'freetype2',
-            default_include_dirs=[
-                'include/freetype2', 'freetype2',
-                'lib/freetype2/include',
-                'lib/freetype2/include/freetype2'],
-            default_library_dirs=[
-                'freetype2/lib'],
-            default_libraries=['freetype', 'z'])
+        if options.get('local_freetype'):
+            src_path = os.path.join(
+                'build', 'freetype-{0}'.format(LOCAL_FREETYPE_VERSION))
+            # Statically link to the locally-built freetype.
+            # This is certainly broken on Windows.
+            ext.include_dirs.insert(0, os.path.join(src_path, 'include'))
+            ext.extra_objects.insert(
+                0, os.path.join(src_path, 'objs', '.libs', 'libfreetype.a'))
+            ext.define_macros.append(('FREETYPE_BUILD_TYPE', 'local'))
+        else:
+            pkg_config.setup_extension(
+                ext, 'freetype2',
+                default_include_dirs=[
+                    'include/freetype2', 'freetype2',
+                    'lib/freetype2/include',
+                    'lib/freetype2/include/freetype2'],
+                default_library_dirs=[
+                    'freetype2/lib'],
+                default_libraries=['freetype', 'z'])
+            ext.define_macros.append(('FREETYPE_BUILD_TYPE', 'system'))
+
+    def do_custom_build(self):
+        # We're using a system freetype
+        if not options.get('local_freetype'):
+            return
+
+        src_path = os.path.join(
+            'build', 'freetype-{0}'.format(LOCAL_FREETYPE_VERSION))
+
+        # We've already built freetype
+        if os.path.isfile(os.path.join(src_path, 'objs', '.libs', 'libfreetype.a')):
+            return
+
+        tarball = 'freetype-{0}.tar.gz'.format(LOCAL_FREETYPE_VERSION)
+        tarball_path = os.path.join('build', tarball)
+        if not os.path.isfile(tarball_path):
+            tarball_url = 'http://download.savannah.gnu.org/releases/freetype/{0}'.format(tarball)
+
+            print("Downloading {0}".format(tarball_url))
+            if sys.version_info[0] == 2:
+                from urllib import urlretrieve
+            else:
+                from urllib.request import urlretrieve
+
+            urlretrieve(tarball_url, tarball_path)
+
+            if get_file_hash(tarball_path) != LOCAL_FREETYPE_HASH:
+                raise IOError("{0} does not match expected hash.".format(tarball))
+
+        print("Building {0}".format(tarball))
+        cflags = 'CFLAGS="{0} -fPIC" '.format(os.environ.get('CFLAGS', ''))
+
+        subprocess.check_call(
+            ['tar zxf {0}'.format(tarball)], shell=True, cwd='build')
```

Comment:
I don't think that's true. `subprocess.check_call(['tar', 'zxf', tarball], cwd='build')` works just fine for me. Maybe it's because you're passing a string in a list instead of a plain string, or a list of strings?